src/Proto.Cluster/Partition/PartitionIdentityActor.cs

```diff
@@ -206,78 +253,72 @@ private async Task OnActivationRequest(ActivationRequest msg, IContext context)
             //but still within the actors sequential execution
             //but other messages could have been processed in between
 
+            if (_config.DeveloperLogging)
+                Console.Write("S"); //spawned
             //Await SpawningProcess
             context.ReenterAfter(
                 res,
-                rst => {
-
-
-                    var response = res.Result;
-
-                    //TODO: as this is async, there might come in multiple ActivationRequests asking for this
-                    //Identity, causing multiple activations
-
-                    //Check if exist in current partition dictionary
-                    //This is necessary to avoid race condition during partition map transfer.
-                    if (_partitionLookup.TryGetValue(msg.ClusterIdentity, out pid))
-                    {
-                        _spawns.Remove(msg.ClusterIdentity);
-                        context.Respond(new ActivationResponse {Pid = pid});
-                        return Task.CompletedTask;
-                    }
-
-                    //Check if process is faulted
-                    if (rst.IsFaulted)
-                    {
-                        _spawns.Remove(msg.ClusterIdentity);
-                        context.Respond(response);
-                        return Task.CompletedTask;
-                    }
-                    if (response == null)
-                    {
-                        _spawns.Remove(msg.ClusterIdentity);
-                        // context.Respond(new ActivationResponse()
-                        // {
-                        //     
-                        // });
-                        //TODO what do we do in this case?
-                        return Task.CompletedTask;
-                    }
-
-                    _partitionLookup[msg.ClusterIdentity] = response.Pid;
-                    context.Respond(response);
-
+                async rst => {
                     try
                     {
-                        _spawns.Remove(msg.ClusterIdentity);
+                        var response = await rst;
```

Comment:
use proper async await for activation callbacks. also use new Failed setting for failed responses